Documentation/ApiOverview/CachingFramework/Developer/Index.rst

```diff
@@ -1,143 +1,165 @@
-   It is recommended to set up a cache configuration with sane defaults,
-   but administrators should always be able to overwrite them for whatever reason.
+..  code-block:: php
+    :caption: EXT:some_extension/ext_localconf.php
 
-.. code-block:: php
+    // use \TYPO3\CMS\Core\Cache\Backend\TransientMemoryBackend;
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations']['myext_mycache']
+        ??= [];
+    $GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations']['myext_mycache']['backend']
+        ??= TransientMemoryBackend::class;
 
-   // use \TYPO3\CMS\Core\Cache\Backend\TransientMemoryBackend;
-   $GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations']['myext_mycache'] ??= [];
-   $GLOBALS['TYPO3_CONF_VARS']['SYS']['caching']['cacheConfigurations']['myext_mycache']['backend'] ??= TransientMemoryBackend::class;
 
-
-.. _caching-developer-example:
-.. _caching-developer-access:
+..  _caching-developer-example:
+..  _caching-developer-access:
 
 Using the Cache
 ===============
 
-First, we need to prepare injection of our cache by setting it up as service in the :ref:`container
-service configuration <configure-dependency-injection-in-extensions>`:
+First, we need to prepare injection of our cache by setting it up as service in
+the :ref:`container service configuration
+<configure-dependency-injection-in-extensions>`:
 
-.. code-block:: yaml
+..  code-block:: yaml
+    :caption: EXT:some_extension/Configuration/Services.yaml
 
     services:
+      _defaults:
+        autowire: true
+        autoconfigure: true
+        public: false
+
+      Vendor\SomeExtension\:
+        resource: '../Classes/*'
+
       cache.myext_mycache:
         class: TYPO3\CMS\Core\Cache\Frontend\FrontendInterface
         factory: ['@TYPO3\CMS\Core\Cache\CacheManager', 'getCache']
         arguments: ['myext_mycache']
 
-The name of the service for the injection configuration is :`cache.myext_mycache`, the name of the cache is `myext_mycache`.
-Both can be anything you like, just make sure they are unique and clearly hint at the purpose of your cache.
+The name of the service for the injection configuration is
+:yaml:`cache.myext_mycache`, the name of the cache is `myext_mycache` (as
+defined in :php:`ext_localconf.php`). Both can be anything you like, just make
+sure they are unique and clearly hint at the purpose of your cache.
 
-.. note::
-
-   Since TYPO3v10, you should prefer :ref:`dependency injection <DependencyInjection>`
-   to retrieve cache frontends whenever possible and no longer use the :php:`CacheManager` directly.
+..  note::
+    Since TYPO3v10, you should prefer :ref:`dependency injection
+    <DependencyInjection>` to retrieve cache frontends whenever possible and no
+    longer use the :php:`CacheManager` directly.
 
 Here is some example code which retrieves the cache via dependency injection:
 
+..  code-block:: php
+    :caption: EXT:some_extension/Classes/MyClass.php
```

Comment:
I would put this code into a file

Reply:
In the next step with constructor property promotion. This is just a clean-up before adjusting for v12.

Reply:
Aah, know why I didn't do it. It is not valid PHP:

```php
$value = /* ... Logic to calculate value ... */;
```

Thought about moving it into methods but that would be overkill.

Reply:
So can we give an example value instead?